network/message/authorization.go

```diff
@@ -18,21 +26,25 @@ type MsgAuthConfig struct {
 	// Type is a func that returns a new instance of message type.
 	Type func() interface{}
 	// Config is the mapping of network channel to list of authorized flow roles.
-	Config map[channels.Channel]flow.RoleList
+	Config map[channels.Channel]ChannelAuthConfig
 }
 
 // EnsureAuthorized checks if the specified role is authorized to send the message on the provided channel and
 // asserts that the message is authorized to be sent on the channel.
 // Expected error returns during normal operations:
 //   - ErrUnauthorizedMessageOnChannel: the channel is not included in the message's list of authorized channels
 //   - ErrUnauthorizedRole: the role is not included in the message's list of authorized roles for the provided channel
-func (m MsgAuthConfig) EnsureAuthorized(role flow.Role, channel channels.Channel) error {
-	authorizedRoles, ok := m.Config[channel]
+func (m MsgAuthConfig) EnsureAuthorized(role flow.Role, channel channels.Channel, isUnicast bool) error {
+	channelAuthConfig, ok := m.Config[channel]
 	if !ok {
 		return ErrUnauthorizedMessageOnChannel
 	}
 
-	if !authorizedRoles.Contains(role) {
+	if isUnicast && !channelAuthConfig.Unicast {
```

Comment:
This does not _read elegantly_ I would prefer to keep it as it is now it reads more naturally.

Reply:
it's a different check. Instead of only preventing broadcast messages on unicast, this would also prevent unicast messages on broadcast (e.g. broadcasting a `ChunkDataResponse` over pubsub. Let's track this in a separate PR